### ISSUE TITLE
[MNG-1041] Classifier-specific dependencies colliding with generic dependencies

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#3610]
**[John Fallows](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ejit)** opened **[MNG-1041](https://issues.apache.org/jira/browse/MNG-1041?redirect=false)** and commented

It seems like there is a collision between dependencies from the same
project but with different classifiers, causing only the main
dependency artifact to be present on the compilation classpath.

project/
api-module/
impl-module/

The api-module generates 2 artifacts, one with classifier "mock".
The impl-module depends on both of these artifiacts, one with "mock"
classifier at "test" scope and the other with no classifier at the
default "compile" scope.

During compilation of the impl-module tests, classes from the
api-module "mock" artifact are not found on the classpath.

It seems as though the "mock" artifact and the regular artifact are
colliding, causing the "mock" artifact to be removed from the test
compilation classpath.

Both the "mock" and regular artifacts from the api-module have type
"jar".  Even when the "mock" type is changed to "zip", the same
problem occurs.

Perhaps dependency classifiers are ignored while constructing the
compilation classpath?

Brett Porter wrote:
I think its because the project you are depending on is in the
reactor, and its probably a bug.

Basically, its trying to use the built version of that project, but
projects are only identified by group:artifactId

---

**Affects:** 2.0-beta-1

**Issue Links:**
- [SUREFIRE-234](https://issues.apache.org/jira/browse/SUREFIRE-234) Test artifacts should support install and deploy
 (_**"is depended upon by"**_)
